### DISCUSSION TITLE
Is the @type property required?

### DISCUSSION THREAD

**Discussion by Code42Cate** [ResearchObject/ro-crate#224] [Q&A]
Hey everyone,

I've been reading the RO-Crate 1.1 specification and was wondering if the `@type` property is required on every entity? I can not find anything clearly stating this, I also looked into the JSON-LD standard. No examples in the specification have an entity without `@type`, but I found some entities in ROHub crates that do not have them. Thank you for your help :)

Cheers, Jonas

**Comment by stain** [comment]
Hi, good question! The `@type` property is not required by JSON-LD itself, but it is convention to include it, particularly when using schema.org. 

Likewise in RO-Crate we have implied the `@type` is to be used on every entity, but has not stated that as an explicit rule. 

I would suggest we modify this for RO-Crate 1.2 to be explicit, so that every entity has at least `@type` and SHOULD have `name`.  If no appropriate type is available, you can use `Thing` (anything) or `WebPage` (if you're pointing to a page).

**Reply by Code42Cate** [reply]
Thanks for the answer:) That's what I thought and implemented, would love to see it be explicitly mentioned in 1.2!

**Comment by stain** [comment]
`@type` requirement now included in RO-Crate 1.2:
https://www.researchobject.org/ro-crate/specification/1.2-DRAFT/metadata.html#common-principles-for-ro-crate-entities
